### ISSUE TITLE
Initialise project directory with git if available and use git name and email for pyproject.toml

### ISSUE THREAD

**Issue by doolio** [pdm-project/pdm#2220]

**Comment by frostming** [comment]
The title description is unclear, and git username and email are already supported.

**Comment by doolio** [comment]
Apologies for being unclear. I meant if `git` is available on a users machine to execute `git init` as part of `pdm init`. Right now `pdm init` just produces a .gitignore file. I think it would be great if it also initialised the project directory as a vcs (git but perhaps others as well) repository.

As for whether git's `user.name` and `user.email` values are supported it does not seem to be if one has defined their `user.email` in separate .gitconfig files via git's `includeif` feature. See below. Perhaps, I'm unusual in doing this.

![image](https://github.com/pdm-project/pdm/assets/13859009/0d449650-3fe5-4670-9a88-a027f68f3d21)

I hope this is clearer. Thanks for your time.

**Comment by doolio** [comment]
> As for whether git's user.name and user.email values are supported it does not seem to be if one has defined their user.email in separate .gitconfig files via git's includeif feature.

But if I execute `git init` inside a project directory I see the `user.email` value I expect which implies it is only available when the directory is a git respository.

![image](https://github.com/pdm-project/pdm/assets/13859009/8d57a64c-957e-479d-b817-dec018cdab2f)

**Comment by pawamoy** [comment]
Well, indeed, you include the config that sets the user.email only when the current directory is a Git repository. I'm not sure to understand what else you would expect?

- cwd is a repo: user.email is set, PDM can use it
- cwd is not a repo: user.email is not set, PDM can't use it

I'm curious: what is the reason for the `includeif.gitdir` on this particular setting (user.email)?

**Comment by doolio** [comment]
> current repo is named github

No, the repo is named pdm-demo but it is within a directory named github. All projects I may work on that are hosted on github have their repositories within this github directory. I have a similar directory for gitlab hosted projects etc. My git configuration 

![image](https://github.com/pdm-project/pdm/assets/13859009/e8f5b9df-6a0f-4959-b6b8-26b8cd500885)

is such that if a project repo is a subdirectory of the github directory then git's user.email is set to the commit email I wish to use for github. Similarly, if the project repo is a subdirectory of the gitlab directory then git's user.email is set to the commit email I wish to use for gitlab etc. The convenience is I don't need to remember to correctly configure git's user.email setting for projects that may be hosted on different forges provided I create the repo in the respective github or gitlab directories. For example, one could use the same strategy to separate work and personal projects and ensure the correct commit username and email is applied.

See [the git scm book](https://git-scm.com/docs/git-config#_conditional_includes) for more details.

**Comment by pawamoy** [comment]
Thanks for the explanation, that's a nice, convenient setup indeed :slightly_smiling_face: 

IMO PDM shouldn't bother with adding an option to run `git init`. It's easy enough to run it yourself :shrug: 

```bash
git init project
pdm init -np project
```

But that's just my opinion!

**Comment by doolio** [comment]
Fair enough. It is something that [`rye`](https://github.com/mitsuhiko/rye) does which I though was a convenient feature.

**Comment by baggiponte** [comment]
I am not sure I understand your request. When I run `pdm init -n`, the default template is filled as below. The authors field is already filled with my git username and email. Is it perhaps because I am using pdm with cookiecutter as an extra dependency?

```toml
[project]
name = "test"
version = "0.1.0"
description = "Default template for PDM package"
authors = [
    {name = "baggiponte", email = "57922983+baggiponte@users.noreply.github.com"},
]
dependencies = []
requires-python = "==3.10.*"
readme = "README.md"
license = {text = "MIT"}

[tool.pdm]
package-type = "application"
```

**Comment by doolio** [comment]
> The authors field is already filled with my git username and email.

Right, but not only if these are set globally. I have these set differently depending on which forge (GitHub, GitLab etc.) the project I'm working on is hosted.
